solidity/contracts/TokenStaking.sol

```diff
@@ -259,6 +269,117 @@ contract TokenStaking is StakeDelegatable {
         return operators[_operator].packedParams.unpack();
     }
 
+    /**
+     * @dev Locks given operator stake from undelegating for the specified duration.
+     * Only previously authorized operator contract can lock the stake.
+     * @param operator Operator address.
+     * @param duration Lock duration in seconds.
+     */
+    function lockStake(
+        address operator,
+        uint256 duration
+    ) public onlyApprovedOperatorContract(msg.sender) {
+        require(
+            isAuthorizedForOperator(operator, msg.sender),
+            "Not authorized"
+        );
+
+        (, uint256 createdAt, uint256 undelegatedAt) =
+            operators[operator].packedParams.unpack();
+
+        require(
+            block.timestamp > createdAt.add(initializationPeriod),
+            "Operator not initialized"
+        );
+        require(
+            (undelegatedAt == 0) || (block.timestamp <= undelegatedAt),
+            "Operator undelegating"
+        );
+
+        operatorLocks[operator].setLock(
+            msg.sender,
+            uint96(block.timestamp.add(duration))
+        );
+    }
+
+    /**
+     * @dev Unlocks given operator stake. Only previously authorized operator
+     * contract can unlock the stake.
+     * @param operator Operator address.
+     */
+    function unlockStake(
+        address operator
+    ) public {
```

Comment:
Unlocking stake can't harm the operator or interfere with other operator contracts. Realistically we could get away with no ACL here whatsoever, but `isAuthorizedForOperator` implies that the contract must have been able to place a lock on the operator sometime in the past, thus sanity-checking out cases where someone other than an operator contract tries to unlock stake. We don't check if a lock is present because it might have been gotten removed with `releaseExpiredLock`.